### DISCUSSION TITLE
How do I specify a git commit hash as the image tag during hybrid cloud CI/CD?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18367] [Q&A] [answered]
Hey all!
I'm trying to move to a Hybrid Cloud deployment and I'm trying to understand good practices around deploying code/images.

My understanding of running an agent with my custom code is to:
• Use CI/CD to create an image with the latest code I want to run (i.e. the asset definitions etc)
• Specify this image in `dagster_cloud.yaml` 
However, the <https://docs.dagster.io/dagster-cloud/deployment/agents/docker/configuration-reference|example reference> specifies a docker image using the `latest` tag, which is not best practice in Docker as far as I know.
If I wanted to specify e.g. a git commit hash as the docker image tag, is there a mechanism for that?
Basically I want the code the agent uses to update whenever I make a new commit.

I can't set the `dagster_cloud.yaml` to specify the current commit as the tag because I don't know the commit hash before committing, so it's a circular dependency.

I've looked for about an hour in various docs and couldn't find any good examples. Could someone point me in the right direction?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C02LJ7G0LAZ/p1701260334101659?thread_ts=1701260334.101659&cid=C02LJ7G0LAZ).*

**Comment by gibsondan** [comment, marked as answer]
https://github.com/dagster-io/dagster-cloud-hybrid-quickstart/blob/main/.github/workflows/dagster-cloud-deploy.yml has a good example of how to do this. (There's also no particular reason the image tag has to be a git SHA - it can also be a random UUID, the important thing is that it should change on each redeploy).
